src/middlewares/middlewares.ts

```diff
@@ -0,0 +1,20 @@
+import jwt from 'jsonwebtoken'
+
+const localhost = process.env.ORIGIN
+
+const corsOptions = {
+	origin: localhost,
+	credentials: true
+}
+
+const authenticateToken = (req, res, next) => {
+	const {AuthToken} = req.cookies
+	if (AuthToken === null) return res.status(401).send()
+	jwt.verify(AuthToken, process.env.TOKEN_SECRET as string, (err: Error, user: object) => {

```

Comment:
Nie używaj tak typu `object` , nie ma wtedy informacji, jakie pola znajdują się w tym argumencie i wielu przypadkach nie będzie można wykonywać operacji na takim typie. Jeżeli jwt zostanie rozkodowane to wiesz co zostało zakodowane w payload